### DISCUSSION TITLE
2023/04/17 Task

### DISCUSSION THREAD

**Discussion by Samsam-lee** [2P3S/scrum-dice#37] [General]
### 승현
- [디자인 수정사항](https://github.com/orgs/2P3S/discussions/6#discussioncomment-5439925)에서 나왔던 디자인 **fix** 하기
- atomic design pattern - **atoms 요소 구현** ( 전부 다 )
https://github.com/2P3S/scrum-dice/issues/23

### 재순
- npm run docker:dev 를 사용하면 개발 바로 가능
- 패키지를 추가할 일이 있으면 nvm으로 버전 관리해두고 추가하기
- 풀리퀘스트 템플릿에서 특히 why? 에는 이슈를 적어두기!
- PR을 보는 사람 입장에서 생각해보고 작성하기

- 개발할 때 사용가능한 redis를 docker-compose에서!
	- [ ] redis 분리해서 사용 가능하게(server, local)
- 백엔드 -> 소켓
	- [x] Nest JS 로 백엔드 기반 만들기

### 승형
- socket url 
- 클린 아키텍쳐 보고 가능하지 않다면, mvc나 mvvc
- 시퀀스 다이어그램

__issue, pr 제대로 작성하기__

**Comment by hyunnnn98** [comment]
## ✅ 승형 Todo check 

- [x] 시퀀스 다이어그램
   - 최소한의 must api 완료. 
- [x] socket url
- [x] 클린 아키텍쳐 보고 가능하지 않다면, mvc나 mvvc
   - 자체적인 클린아키텍쳐 ❌ , nestjs directory structure 🅾️

### 👀 시퀀스 다이어그램에 대한 FB ...

- 사용자 접속 이력 확인?
   - `parameter` 데이터를 상세하게 적어두기. 인식 맞춤을 위하여. 
-  스크럼 다이스 이력 조회는 어떤 방식으로 이루어지나?
   - `key` 값을 무엇으로? 
- 이력조회에 대한 인식이 서로 틀리다.
   - 과거에 했던 이력은 로그인을 하지 않는 이상 unique한 `key`가 없기에 볼 수 없다.
   - 단지, 매번 생성했던 방에 재접속시 해당 방에 대한 이력은 조회할 수 있다. ( url 경유로 들어가야 조회 가능 )
- 방 생성후 SocketServer 는 방 생성 유저에게 생성된 방 `key` 를 전달해야한다.
- user1, user2의 차이를 명확하게 하자. 

### 👀 socket url 설계에 대한 FB ...

- 시퀀스 다이어그램의 FB을 바탕으로 재구성해보기.

**Comment by JeongJaeSoon** [comment]
## ✅ 승현 Todo Check
* [ ] [디자인 수정사항](https://github.com/orgs/2P3S/discussions/6#discussioncomment-5439925)에서 나왔던 디자인 fix 하기
* [x] atomic design pattern - atoms 요소 구현
	* @Samsam-lee  썜썜이 리뷰 빨리 하셈
	* 대충 보면 안됨!

## ✅  재순 Todo Check
* [ ] Redis 분리
	* AWS 기동 중인것은 일단 정지
	* backend 쪽에서 함께 사용할 수 있게 기반 작성
* [x] Chat app socket.io server 작성
